lib/cosmos/tools/table_manager/table_manager.rb

```diff
@@ -958,112 +786,346 @@ def update_gui_item(table_name, table_def, item_def, table_row, table_column)
     end
 
     # Delete all the tabs in the table manager gui and initialize globals to
     # prepare for the next table to load.
-    def delete_tabs
-      @gui_tables = Hash.new
-      @ordered_gui_table_names = Array.new
-      @currently_displayed_table_name = ""
-
+    def reset_gui
+      set_table_modified(false)
       @tabbook.tabs.each_with_index do |tab, index|
         tab.dispose
         @tabbook.removeTab(index)
       end
     end
 
-    def self.run(option_parser = nil, options = nil)
-      Cosmos.catch_fatal_exception do
-        unless option_parser and options
-          option_parser, options = create_default_options()
-          options.width = 800
-          options.height = 600
-          options.title = "Table Manager"
-          options.auto_size = false
-          options.no_tables = false
-          option_parser.separator "Table Manager Specific Options:"
-          option_parser.on("-n", "--notables", "Do not include table file editing options. This will remove the 'Table' menu.") do
-            options.no_tables = true
+    def save_all_gui_data
+      result = ''
+      @core.config.tables.each do |table_name, table|
+        begin
+          save_gui_data(table_name)
+        rescue SaveError => err
+          result << "\nErrors in #{table_name}:\n#{err.message}"
+        end
+      end
+      raise SaveError, result.lstrip unless result.empty?
+    end
+
+    def display_all_gui_data
+      reset_gui()
+      @core.config.tables.each do |table_name, table|
+        create_table_tab(table)
+        display_gui_data(table_name)
+      end
+    end
+
+    def display_gui_data(table_name)
+      table = @core.config.table(table_name)
+      gui_table = @tabbook.tab(table_name)
+      return unless table && gui_table
+
+      gui_table.blockSignals(true) # block signals while we programatically update it
+      # Cancel any table selections so the text will be visible when it is refreshed
+      gui_table.clearSelection
+
+      set_table_headers(table, gui_table)
+
+      row = 0
+      column = 0
+      table.sorted_items.each do |item|
+        next if item.hidden
+        update_gui_item(table_name, table, gui_table, item, row, column)
+
+        if table.type == :TWO_DIMENSIONAL
+          # only increment our row when we've processed all the columns
+          if column == table.num_columns - 1
+            row += 1
+            column = 0
+          else
+            column += 1
           end
+        else
+          row += 1
         end
+      end
 
-        super(option_parser, options)
+      gui_table.resizeColumnsToContents()
+      gui_table.resizeRowsToContents()
+      gui_table.blockSignals(false)
+    end
+
+    # Updates the table by setting the value in the table to the properly formatted value.
+    def update_gui_item(table_name, table, gui_table, item, row, column)
+      value = table.read(item.name, :FORMATTED)
+
+      if item.states && item.states.keys.sort == %w(CHECKED UNCHECKED)
```

Comment:
If a parameter has two states called CHECKED and UNCHECKED (values don't matter) then I automatically convert that to a checkbox in the GUI.